src/main/java/org/tarantool/TarantoolClientImpl.java

```diff
@@ -265,33 +289,63 @@ protected void configureThreads(String threadName) {
      * @return deferred result
      */
     protected Future<?> exec(long timeoutMillis, Code code, Object... args) {
-        return doExec(timeoutMillis, code, args);
+        return doExec(timeoutMillis, code, ResolvableArgument.ofAllResolved(args));
     }
 
-    protected TarantoolOp<?> doExec(long timeoutMillis, Code code, Object[] args) {
-        validateArgs(args);
-        long sid = syncId.incrementAndGet();
+    @Override
+    protected int resolveSpace(String space) {
```

Comment:
Do I understand right that if we're currently fetching a new schema (so we're know that a current one if old) and at this time a user fired a new request with a space that should appear with this new schema, the request will fail with TarantoolSpaceNotFoundException? If so, it worth to mention in the README. Say, a user want to do DDL and then DML request. Should it wait for the first request or can do the second w/o waiting?

Reply:
Not exactly, if a client knows that it has the outdated schema it will try to execute a delayed operation after. 

> Should it wait for the first request or can do the second w/o waiting?
No, a client guarantees operations are executed in the order they were registered in the client.

Reply:
Now we using the approach with ping request and decided to don't provide ordering guarantees, so this discussion thread does not look as actual.